Bug 1955822: Ingress: Mount router stats secret as a volume

sgreene570 opened the pull request:
openshift/router#291 gave the router
the ability to use stats credentials contained in separate username
and password files for security reasons.
This commit modifies the Ingress deployment so that the router-stats
secret is mounted into router pods, and the filenames for the username
and password files are passed to the Router binary via new
environment variables.

pkg/operator/controller/ingress/deployment.go:

Add a new volume and volume mount for the router stats secret.
Replace the STATS_USERNAME and STATS_PASSWORD environment variables with
the new STATS_USER_NAME_FILE and STATS_PASSWORD_FILE environment
variables.

pkg/operator/controller/ingress/deployment_test.go:

Update `TestDeploymentConfigChanged` to reflect the new stats-auth
volume.

Update `TestDesiredRouterDeployment` to verify that the expected
volumes exist with the proper secret references.

This commit is in support of Bug 1955822.

sgreene570 commented:
/hold for https://github.com/openshift/router/pull/291

sgreene570 commented:
/hold for https://github.com/openshift/router/pull/291

sgreene570 commented:
/test unit

Miciah commented on pkg/operator/controller/ingress/deployment.go:
```diff
@@ -358,24 +358,28 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
-				Key: "statsPassword",
-			},
-		}},
+		{Name: "STATS_USERNAME_FILE", Value: statsVolumeMountPath + "/statsUsername"},
+		{Name: "STATS_PASSWORD_FILE", Value: statsVolumeMountPath + "/statsPassword"},
```
```suggestion
		{Name: "STATS_USERNAME_FILE", Value: filepath.Join(statsVolumeMountPath, "statsUsername")},
		{Name: "STATS_PASSWORD_FILE", Value: filepath.Join(statsVolumeMountPath, "statsPassword")},
```

sgreene570 replied:
fixed. Thanks!

Miciah reviewed (commented):
Looks good!  `TestDesiredRouterDeployment` probably should check for the existence of the "stats-auth" volume or volume mount and the `STATS_USERNAME_FILE` and `STATUS_PASSWORD_FILE` environment variables.

sgreene570 commented:
> Looks good! `TestDesiredRouterDeployment` probably should check for the existence of the "stats-auth" volume or volume mount and the `STATS_USERNAME_FILE` and `STATUS_PASSWORD_FILE` environment variables.

Fixed! I also refactored some of the existing volume checks. Thanks!

sgreene570 commented:
https://github.com/openshift/router/pull/291 has merged.
/hold cancel

sgreene570 commented:
/retest

Miciah commented:
Thanks!
/lgtm
